### DISCUSSION TITLE
how to add duckdb jdbc driver to h2 database?

### DISCUSSION THREAD

**Discussion by l1t1** [duckdb/duckdb#6525] [Q&A]
h2 documents says
>Adding Database Drivers
To register additional JDBC drivers (MySQL, PostgreSQL, HSQLDB,...), add
the jar file names to the environment variables H2DRIVERS or CLASSPATH.
Example (Windows): to add the HSQLDB JDBC driver C:\Programs\hsqldb\
lib\hsqldb.jar, set the environment variable H2DRIVERS to C:\Programs\
hsqldb\lib\hsqldb.jar.
Multiple drivers can be set; entries need to be separated by ; (Windows) or
: (other operating systems). Spaces in the path names are supported. The
settings must not be quoted.

but I cannot find the driver install steps at the intallation page of duckdb documnet 
it says
```
Installation
<dependency>
 <groupId>org.duckdb</groupId>
 <artifactId>duckdb_jdbc</artifactId>
 <version>0.7.1</version>
</dependency>
Usage Example
Class.forName("org.duckdb.DuckDBDriver");
Connection conn = DriverManager.getConnection("jdbc:duckdb:");
Statement stmt = conn.createStatement();
ResultSet rs = stmt.executeQuery("SELECT 42");
```
I also found the file in release page
https://github.com/duckdb/duckdb/releases/download/v0.7.1/duckdb_jdbc-windows-amd64.jar
it isn't documented

**Comment by l1t1** [comment]
I copied the [duckdb_jdbc-windows-amd64.jar](https://github.com/duckdb/duckdb/releases/download/v0.7.1/duckdb_jdbc-windows-amd64.jar) to d:\duckdb, and add d:\duckdb\duckdb_jdbc-windows-amd64.jar to H2DRIVERS and CLASSPATH environment variables
how to fill the setting fields?
![2023-03-02_100550](https://user-images.githubusercontent.com/23301704/222312257-99a3c5ac-ef50-4e12-a9b8-a6f66e3f0b65.png)

**Reply by Mause** [reply]
Usually people download JDBC drivers from maven 

Have a look at the docs for the driver class and connection string: https://duckdb.org/docs/api/java

**Reply by l1t1** [reply]
it works using following strings according the API docs, thanks,
org.duckdb.DuckDBDriver
jdbc:duckdb:
![2023-03-02_105849](https://user-images.githubusercontent.com/23301704/222320041-85784db7-7a8a-48bd-9eb4-41eda898b7a4.png)

**Comment by l1t1** [comment]
it seems that the duckdb driver has a question, the information of datbase is not right (the left tree)
I use  the same method to connect sqlite(https://repo1.maven.org/maven2/org/xerial/sqlite-jdbc/3.41.0.0/sqlite-jdbc-3.41.0.0.jar), it's correct
![2023-03-02_122439](https://user-images.githubusercontent.com/23301704/222331174-8b46a959-74ef-44e7-9c78-eda9f9f3505c.png)
